providers/dns/constellix/constellix.go

```diff
@@ -263,6 +269,12 @@ func containsValue(record *internal.Record, value string) bool {
 	return false
 }
 
-func getRecordName(fqdn, authZone string) string {
-	return fqdn[0 : len(fqdn)-len(authZone)-1]
+func getRecordName(fqdn, authZone string) (string, error) {
+	end := len(fqdn) - len(authZone) - 1
+
+	if len(fqdn) < end || end < 0 {
+		return "", fmt.Errorf("%d is lower than the length of the fqdn (fqdn: %s, authZone: %s)", end, fqdn, authZone)
+	}
+
+	return fqdn[0:end], nil
```

Comment:
This function is repeated four times. Can we move it into an utilities package? `dns/providers/internal` already exists, so having a "records" sub package there wouldn't hurt that much.

Reply:
The name is generic, we have several different implementations, so I don't want to create confusion.

Reply:
I didn't know we had a "generic" package. For me, the important bit is that there are four copy-pasted implementations of non-trivial behaviour without tests.

Reply:
Q: Is it possible to have `fqdn == "foo.example.com."` and `authZone == "example.org"`? 

If so, this function returns `"foo", nil`, and I'm not sure if this is another bug lingering here.

Reply:
Ah, I believe I can answer that myself: `authZone` is the domain containing the SOA record for `fqdn`. IIRC, this requires `fqdn` to be a sub-domain of `authZone`.

Reply:
I'm not a huge fan of "utils" package. I have to think.

Reply:
I created a function `ExtractSubDomain` inside the `dns01` package.